src/test/scala/com/cognite/spark/datasource/CdpRddTest.scala

```diff
@@ -61,85 +73,109 @@ class CdpRddTest extends FlatSpec with Matchers with SparkTest {
       val expectedCursors: Iterator[Option[String]] =
         Iterator(Option.empty[String]) ++ Stream.from(0).map(n => Some(n.toString)).toIterator
       val numberedItems = NumberedItems(Stream.from(0).map(_.toString).toIterator)
-      override def getJson[A: Decoder](auth: Auth, url: Uri, maxRetries: StatusCode): IO[A] = {
+      override def getJson[A: Decoder](
+          auth: Auth,
+          url: Uri,
+          maxRetries: StatusCode,
+          applicationId: Option[String]
+      ): IO[A] = {
         url.paramsMap.get("limit") should be(Some("2"))
         url.paramsMap.get("cursor") should be(expectedCursors.next)
         IO.pure(decode(numberedItems.next).right.get)
       }
     }
 
-    rdd.partitions should have length 5
-    rdd.partitions.map(p => p.asInstanceOf[CdpRddPartition].cursor) should
-      contain only (Seq(None) ++ (0 to 3).map(x => Some(x.toString)): _*)
+    (rdd.partitions should have).length(5)
```

Comment:
While I'm not the biggest fan, ScalaTest does have a DSL and we (as well as most others) use the DSLish way of writing our tests. Don't change it in this PR. If we want to change it we should do so in a separate PR and we should change all tests at once, but I don't think we should tbh.